agent/api/container.go

```diff
@@ -107,7 +107,6 @@ type Container struct {
 	// setter/getter.  When this is done, we need to ensure that the UnmarshalJSON is
 	// handled properly so that the state storage continues to work.
 	SentStatusUnsafe ContainerStatus `json:"SentStatus"`
-	sentStatusLock   sync.RWMutex
```

Comment:
This can't be the extent of this. This lock is used across many places in the code. Please modify all of them. Probably why build is broken.